hack/e2e.sh

```diff
@@ -21,3 +21,17 @@ if [ -z "$SUITE" ] || [ "$SUITE" == "clusterreader" ]; then
   (awk '/^  azureClusterReaderKubeconfig:/ { print $2 }' <_data/containerservice.yaml | base64 -d) > _data/_out/azure-cluster-reader.kubeconfig
   go test ./test/e2e -test.v -ginkgo.v -ginkgo.focus="\[AzureClusterReader\]" -ginkgo.noColor -ginkgo.randomizeAllSpecs -tags e2e -kubeconfig ../../_data/_out/azure-cluster-reader.kubeconfig 2>&1 | tee azure-reader.log
 fi
+
+if [ -z "$SUITE" ] || [ "$SUITE" == "dedicated-cluster-admin" ]; then
+  echo "Running azure dedicated-cluster-admin tests"
+  fqdn=$(awk '/^  fqdn:/ { print $2 }' <_data/containerservice.yaml)
+  export KUBECONFIG_DEDICATED_ADMIN=_data/_out/dedicated-cluster-admin.kubeconfig
+  export KUBECONFIG_DEDICATED_READER=_data/_out/dedicated-cluster-reader.kubeconfig
+  export KUBECONFIG_NORMAL_USER=_data/_out/normal.kubeconfig
+
+  # oc login is going to create the kubeconfig with the proper permissions
+  KUBECONFIG=$KUBECONFIG_DEDICATED_ADMIN oc login $fqdn --username dedicated-cluster-admin --password "$(awk '/^  dedicatedAdminPasswd:/{ print $2 }' <_data/containerservice.yaml)" --insecure-skip-tls-verify=true
+  KUBECONFIG=$KUBECONFIG_DEDICATED_READER oc login $fqdn --username dedicated-cluster-reader --password "$(awk '/^  dedicatedReaderPasswd:/{ print $2 }' <_data/containerservice.yaml)" --insecure-skip-tls-verify=true
+  KUBECONFIG=$KUBECONFIG_NORMAL_USER oc login $fqdn --username normal --password "$(awk '/^  normalUserPasswd:/{ print $2 }' <_data/containerservice.yaml)" --insecure-skip-tls-verify=true
```

Comment:
Why do you need this?

Reply:
You probably only need the dedicated-admin kubeconfig so remove the rest of the logins?

Reply:
We wanted to be able to run tests as each user.  The login produces the kubeconfig for each user and then in the e2e we will create a client per user.  That was the intention.

Reply:
Then you need three different invocations of `go test`. Also, isn't normal user already being tested above?

Reply:
I think it was @jim-minter's idea to have a single invocation of the testing framework and pass in the path to the kubeconfigs.  I'm fine invoking the test script once per user.

We could piggy back on the first e2e section to enhance the user's tests or place them here.  I'm fine either way.  I think this is here to verify we didn't grant the `normal|enduser` escalated dedicated privileges.

Reply:
Hmm, thinking about this we might need more than 1 kubeconfig inside a single test as a normal user will need to create objects and then the admin and reader will need to look at those objects.

Reply:
Can you walk me through an example where you need that and you cannot mock the creation of the objects for dedicated admins/readers?

Reply:
Btw changing the framework to use multiple kubeconfigs is a big change and probably not necessary for this PR so let's take this discussion in a separate issue.

Reply:
The scenario we would like to test is that a normal user creates a project and creates a pod.  The dedicated-cluster-admin user should have access to manage the project that the normal user created.  

By using the same kubeconfig, if the dedicated-cluster-admin creates a project he will be added to the admin rolebinding for the project by default and therefore be able to manage the project.  This fails to test that the clusterrole dedicated-cluster-admin has access to the project.

Hopefully this scenario makes sense.

Reply:
Yes, it makes sense, thanks for the explanation. Can you open a follow-up issue?

Reply:
Nevermind about the follow-up issue, I see you already create a namespace as a normal user and then read/write as a dedicated admin.